### ISSUE TITLE
C++ Support

### ISSUE THREAD

**Issue by reesdraminski** [umbchackers/hardModeCode#8]
For CodeMirror Syntax highlighting you need the [C-like](https://codemirror.net/mode/clike/index.html) mode. Compile at C++11 and could test with [Google Test](https://github.com/google/googletest).

**Comment by Gim949** [comment]
Do we need to use Google Test here? I figured that since, at its core, we're comparing outputs based on the  correct code versus the user's code, all we would really need to do is just compile and run the user's code and then compare outputs too see if we get what we expected. It might not be as clean as running our own tests against their code but it does keep all of our problems in one code base (we wouldn't have to write up the same problem in 3+ different languages). 

This would probably mean that we would have to write up our own testing framework though...

**Comment by reesdraminski** [comment]
Yeah you're right, having a test framework for each language seems like overkill and would lead to too much code/dependencies long-term. I do like the idea of a custom "framework," since we only would really need the equivalent to AssertEqual. I enjoy the idea of parallelizing tests to make it run fast, but I don't think any of our problems would have enough computational overhead to warrant that though.

We would need to automatically append some function calls to test the user's code before we save the file to disk, but those functions could be doing something as simple as printing the output of our test functions, and then we would do string comparison in Node.
